doc/languages-frameworks/maven.md

```diff
@@ -0,0 +1,351 @@
+
+We make sure to provide this classpath to the _makeWrapper_.
+
+> readTree may be soon availble in nixpkgs via https://github.com/NixOS/nixpkgs/pull/101096
+
+```nix
+{ pkgs ? import <nixpkgs> { }, stdenv ? pkgs.stdenv, lib ? pkgs.lib
+, maven ? pkgs.maven, callPackage ? pkgs.callPackage
+, makeWrapper ? pkgs.makeWrapper, jre ? pkgs.jre }:
+let
+  readTree = dir:
+    lib.flatten (lib.mapAttrsToList (name: type:
+      if (type == "directory") then
+        readTree (dir + "/${name}")
+      else
+        dir + "/${name}") (builtins.readDir dir));
+
+  repository = callPackage ./build-maven-repository.nix { };
+  jars = lib.traceVal
+    (builtins.filter (path: (lib.hasSuffix ".jar" (toString path)))
+      (readTree repository));
+  classpath = lib.concatStringsSep ":" jars;
+in stdenv.mkDerivation rec {
+  pname = "maven-demo";
+  version = "1.0";
+
+  src = builtins.fetchTarball
+    "https://github.com/fzakaria/nixos-maven-example/archive/main.tar.gz";
+  buildInputs = [ maven makeWrapper ];
+
+  buildPhase = ''
+    echo "Using repository ${repository}"
+    mvn --offline -Dmaven.repo.local=${repository} package;
+  '';
+
+  installPhase = ''
+    mkdir -p $out/bin
+    mkdir -p $out/share/java
+
+    install -Dm644 target/${pname}-${version}.jar $out/share/java
+    # create a wrapper that will automatically set the classpath
+    # this should be the paths from the dependency derivation
+    makeWrapper ${jre}/bin/java $out/bin/${pname} \
+          --add-flags "-classpath $out/share/java/${pname}-${version}.jar:${classpath}" \
+          --add-flags "Main"
+  '';
+}
+```
+
+### MANIFEST file via Maven Plugin
+
+Augment the _pom.xml_ to create a JAR with the following manifest:
+```xml
+<build>
+  <plugins>
+    <plugin>
+        <artifactId>maven-jar-plugin</artifactId>
+        <configuration>
+            <archive>
+                <manifest>
+                    <addClasspath>true</addClasspath>
+                    <classpathPrefix>../../repository/</classpathPrefix>
+                    <classpathLayoutType>repository</classpathLayoutType>
+                    <mainClass>Main</mainClass>
+                </manifest>
+                <manifestEntries>
+                    <Class-Path>.</Class-Path>
+                </manifestEntries>
+            </archive>
+        </configuration>
+    </plugin>
+  </plugins>
+</build>
+```
+
+> This instructs the JAR to look for the necessary dependencies in the _lib/_ relative folder. The layout of the folder is also in the _maven repository_ style.
+
+We will modify the derivation above to add a symlink to our repository so that it's accessible to our JAR during the _installPhase_.
+
+```nix
+{ pkgs ? import <nixpkgs> { }, stdenv ? pkgs.stdenv, lib ? pkgs.lib
+, maven ? pkgs.maven, callPackage ? pkgs.callPackage
+, makeWrapper ? pkgs.makeWrapper, jre ? pkgs.jre }:
```

Comment:
These defaults exist because for the purpose of demonstration each file (https://github.com/fzakaria/nixos-maven-example) is individually runnable via _nix-build_